File: 10409/main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int top, north, west;

void init(){
    top = 1;
    north = 2;
    west = 3;
}

void rotate_a(int *a){
    int tmp = top;
    top = 7 - *a;
    *a = tmp;
}

void rotate_b(int *a){
    int tmp = top;
    top = *a;
    *a = 7 - tmp;
}

void solve(int n){
    char s[6];
    for(; n; n--){
        scanf("%s", s);
        if(!strcmp(s, "north")) rotate_a(&north);
        if(!strcmp(s, "west"))  rotate_a(&west);
        if(!strcmp(s, "south")) rotate_b(&north);
        if(!strcmp(s, "east"))  rotate_b(&west);
    }
}

int main()
{
    int db;
    while(scanf("%d", &db) && db)
    {
        init();
        solve(db);
        printf("%d\n", top);
    }
    return 0;
}

File: 10409/Main.java
import java.util.Scanner;

public class Main {
    static Scanner sc = new Scanner(System.in);
    public static void main(String[] args) {
        int rolls, top, north, west, temporary;
        String command;
        while((rolls = Integer.parseInt(sc.nextLine())) != 0){
            top = 1;
            north = 2;
            west = 3;

            while(rolls > 0){
                command = sc.nextLine();

                if(command.equals("north")){
                    temporary = top;
                    top = 7 - north;
                    north = temporary;
                }

                if(command.equals("west")){
                    temporary = top;
                    top = 7 - west;
                    west= temporary;
                }

                if(command.equals("south")){
                    temporary = top;
                    top = north;
                    north = 7 - temporary;
                }

                if(command.equals("east")){
                    temporary = top;
                    top = west;
                    west = 7 - temporary;
                }

                rolls--;
            }

            System.out.println(top);
        }
    }
}
